linkerd/app/outbound/src/http.rs

```diff
@@ -1,164 +1,142 @@
-mod concrete;
-pub mod detect;
-mod endpoint;
-pub mod logical;
-mod proxy_connection_close;
-mod require_id_header;
-mod retry;
-mod server;
-mod strip_proxy_error;
-
 use self::{
     proxy_connection_close::ProxyConnectionClose, require_id_header::NewRequireIdentity,
     strip_proxy_error::NewStripProxyError,
 };
-pub(crate) use self::{require_id_header::IdentityRequired, server::ServerRescue};
-use crate::tcp;
-pub use linkerd_app_core::proxy::http::*;
+use crate::{tcp, Outbound};
 use linkerd_app_core::{
-    profiles::{self, LogicalAddr},
-    proxy::{api_resolve::ProtocolHint, tap},
-    svc, tls, Addr, Conditional, CANONICAL_DST_HEADER,
+    io, profiles,
+    proxy::{
+        api_resolve::{ConcreteAddr, Metadata},
+        core::Resolve,
+    },
+    svc,
+    transport::addrs::*,
+    Addr, Error,
 };
-use std::{net::SocketAddr, str::FromStr};
-
-pub type Logical = crate::logical::Logical<Version>;
-pub type Concrete = crate::logical::Concrete<Version>;
-pub type Endpoint = crate::endpoint::Endpoint<Version>;
+use std::{fmt::Debug, hash::Hash};
 
-pub type Connect = self::endpoint::Connect<Endpoint>;
+mod concrete;
+mod endpoint;
+mod logical;
+mod proxy_connection_close;
+mod require_id_header;
+mod retry;
+mod server;
+mod strip_proxy_error;
 
-#[derive(Clone, Debug)]
-pub struct CanonicalDstHeader(pub Addr);
+pub use self::logical::Logical;
+pub(crate) use self::require_id_header::IdentityRequired;
+pub use linkerd_app_core::proxy::http::{self as http, *};
 
-// === impl CanonicalDstHeader ===
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct Http {
+    target: Logical,
+    version: http::Version,
+}
 
-impl From<CanonicalDstHeader> for HeaderPair {
-    fn from(CanonicalDstHeader(dst): CanonicalDstHeader) -> HeaderPair {
-        HeaderPair(
-            HeaderName::from_static(CANONICAL_DST_HEADER),
-            HeaderValue::from_str(&dst.to_string()).expect("addr must be a valid header"),
-        )
+// === impl Outbound ===
+
+impl<C> Outbound<C> {
+    /// Builds a stack that handles protocol detection as well as routing and
+    /// load balancing for a single logical destination.
```

Comment:
copyedit nit, not important: maybe
```suggestion
    /// Builds a stack that handles protocol detection, routing, and
    /// load balancing for a single logical destination.
```